quic/s2n-quic-transport/src/stream/api.rs

```diff
@@ -53,8 +53,8 @@ impl State {
 
 impl Drop for State {
     fn drop(&mut self) {
-        let is_rx_open = self.rx.is_open();
-        let is_tx_open = self.tx.is_open();
+        let is_rx_open = !self.rx.is_closed();
+        let is_tx_open = !self.tx.is_closed();
```

Comment:
why is this necessary?

Reply:
`is_open` is only true if the status is set to `Open` whereas `is_closed` is only true if the status is `Finished` or `Reset`. The latter is what we want here since we want to detach in any state outside of the completely terminal ones.